### ISSUE TITLE
Initial patches sync: Detected a bad mac

### ISSUE THREAD

**Issue by loveGit99** [Auties00/WhatsappWeb4j#203]
// codes:
Whatsapp.newConnection()
                .addLoggedInListener((api) -> {
                    System.out.println("Connected!");
                        try {
                            Chat chat = Chat.ofJid(ContactJid.of("8615xxxxx5202"));
                            CompletableFuture<MessageInfo> msgFutrue = api.sendMessage(chat, "Hi!");
                            MessageInfo msgInfo = msgFutrue.get(1, TimeUnit.MINUTES);
                            Thread.sleep(5000);
                        } catch (Exception e) {
                            throw new RuntimeException(e);
                        }
                })
                .addNewMessageListener((whatsapp, info) -> whatsapp.sendMessage(info.chatJid(), "Automatic answer", info))
                .connect()
                .join();

//  output:

Connected!
2023 9:55:00  it.auties.whatsapp.api.ErrorHandler lambda$defaultErrorHandler$1
Fatal: Socket failure at MEDIA_CONNECTION
java.util.concurrent.CompletionException: it.auties.whatsapp.exception.ErroneousNodeRequestException: Node timed out(Node[description=iq, attributes={xmlns=w:m, to=s.whatsapp.net, id=cb-2, type=set}, content=[Node[description=media_conn]]]), no response from WhatsApp
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:636)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
	at it.auties.whatsapp.model.request.Request.cancelTimedFuture(Request.java:79)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
	at java.base/java.util.concurrent.ForkJoinTask.doExec$$$capture(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: it.auties.whatsapp.exception.ErroneousNodeRequestException: Node timed out(Node[description=iq, attributes={xmlns=w:m, to=s.whatsapp.net, id=cb-2, type=set}, content=[Node[description=media_conn]]]), no response from WhatsApp
	at it.auties.whatsapp.model.request.Request.cancelTimedFuture(Request.java:75)
	... 7 more
Caused by: java.lang.RuntimeException
	at it.auties.whatsapp.model.request.Request.<init>(Request.java:47)
	at it.auties.whatsapp.model.request.Request.of(Request.java:55)
	at it.auties.whatsapp.model.request.Node.toRequest(Node.java:308)
	at it.auties.whatsapp.socket.SocketHandler.send(SocketHandler.java:307)
	at it.auties.whatsapp.socket.SocketHandler.sendQuery(SocketHandler.java:379)
	at it.auties.whatsapp.socket.SocketHandler.sendQuery(SocketHandler.java:351)
	at it.auties.whatsapp.socket.StreamHandler.createMediaConnection(StreamHandler.java:655)
	at it.auties.whatsapp.socket.StreamHandler.digestSuccess(StreamHandler.java:497)
	at it.auties.whatsapp.socket.StreamHandler.digest(StreamHandler.java:109)
	at it.auties.whatsapp.socket.SocketHandler.handleNode(SocketHandler.java:227)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at it.auties.whatsapp.socket.SocketHandler.onBinary(SocketHandler.java:221)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.glassfish.tyrus.core.AnnotatedEndpoint.callMethod(AnnotatedEndpoint.java:556)
	at org.glassfish.tyrus.core.AnnotatedEndpoint$WholeHandler$1.onMessage(AnnotatedEndpoint.java:655)
	at org.glassfish.tyrus.core.TyrusSession.notifyMessageHandlers(TyrusSession.java:591)
	at org.glassfish.tyrus.core.TyrusEndpointWrapper.onMessage(TyrusEndpointWrapper.java:806)
	at org.glassfish.tyrus.core.TyrusWebSocket.onMessage(TyrusWebSocket.java:183)
	at org.glassfish.tyrus.core.frame.BinaryFrame.respond(BinaryFrame.java:70)
	at org.glassfish.tyrus.core.ProtocolHandler.process(ProtocolHandler.java:819)
	at org.glassfish.tyrus.client.TyrusClientEngine$TyrusReadHandler.handle(TyrusClientEngine.java:726)
	at org.glassfish.tyrus.container.grizzly.client.GrizzlyClientFilter$ProcessTask.execute(GrizzlyClientFilter.java:452)
	at org.glassfish.tyrus.container.grizzly.client.TaskProcessor.processTask(TaskProcessor.java:91)
	at org.glassfish.tyrus.container.grizzly.client.TaskProcessor.processTask(TaskProcessor.java:68)
	at org.glassfish.tyrus.container.grizzly.client.GrizzlyClientFilter.handleRead(GrizzlyClientFilter.java:248)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:88)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:246)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:178)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:118)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:96)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:51)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:510)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:82)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:83)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:101)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:535)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:515)
	at java.base/java.lang.Thread.run(Thread.java:833)

**Comment by Auties00** [comment]
What version are you using?

**Comment by loveGit99** [comment]
> What version are you using?

       <dependency>
            <groupId>com.github.auties00</groupId>
            <artifactId>whatsappweb4j</artifactId>
            <version>3.0.2</version>
        </dependency>

add comments:   The phone number  8615xxxxx5202  is not in my contacts.

**Comment by Auties00** [comment]
> > What version are you using?
> 
>        <dependency>
>             <groupId>com.github.auties00</groupId>
>             <artifactId>whatsappweb4j</artifactId>
>             <version>3.0.2</version>
>         </dependency>
> 
> add comments:   The phone number  8615xxxxx5202  is not in my contacts.

The error you sent isn't related to the send message function, does anything else appear. I'm gonna guess that i broke something in 3.0.2 as others are having problems as well btw

**Comment by loveGit99** [comment]
I change the version to 3.0-RC10 , but the application throws the following exception.
> codes:
```xml
        <dependency>
            <groupId>com.github.auties00</groupId>
            <artifactId>whatsappweb4j</artifactId>
            <version>3.0-RC10</version>
        </dependency>
```

 ```java

 Whatsapp api = Whatsapp.newConnection().registerListener(new WhatsappListener() {
            @Override
            public void onLoggedIn() {
                System.out.println("Connected!");
            }
        });
        CompletableFuture<Whatsapp> connFuture = api.connect();
        connFuture.get();
        Chat chat = Chat.ofJid(ContactJid.of("8615919785202@s.whatsapp.net"));
        CompletableFuture<MessageInfo> msgFutrue = api.sendMessage(chat, "Hi!");
        MessageInfo msgInfo = msgFutrue.get(1, TimeUnit.MINUTES);

```

> output
Connected!
 2023 12:17:07 AM  it.auties.whatsapp.binary.BinarySocket$ErrorHandler handleFailure
Warning: Socket failure at MESSAGE(syncType is marked non-null but is null
 at [Source: (byte[])[2 bytes]; byte offset: #2] (through reference chain: it.auties.whatsapp.model.sync.HistorySync$HistorySyncBuilder["syncType"]))
2023 12:17:07 AM  it.auties.whatsapp.binary.BinarySocket$ErrorHandler handleFailure
Warning: Saved stacktrace at: C:\Users\Administrator\.whatsappweb4j\exceptions\1673799427753.txt
2023 12:17:07 AM  it.auties.whatsapp.binary.BinarySocket$ErrorHandler handleFailure
Warning: Restoring session

> ------------1673799427753.txt   content following
com.fasterxml.jackson.databind.JsonMappingException: syncType is marked non-null but is null
 at [Source: (byte[])[2 bytes]; byte offset: #2] (through reference chain: it.auties.whatsapp.model.sync.HistorySync$HistorySyncBuilder["syncType"])
	at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:276)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty._throwAsIOE(SettableBeanProperty.java:623)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty._throwAsIOE(SettableBeanProperty.java:611)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeSetAndReturn(MethodProperty.java:173)
	at com.fasterxml.jackson.databind.deser.BuilderBasedDeserializer.vanillaDeserialize(BuilderBasedDeserializer.java:293)
	at com.fasterxml.jackson.databind.deser.BuilderBasedDeserializer.deserialize(BuilderBasedDeserializer.java:217)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:323)
	at com.fasterxml.jackson.databind.ObjectReader._bindAndClose(ObjectReader.java:2051)
	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1529)
	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1544)
	at it.auties.protobuf.api.jackson.ProtobufMapper.readMessage(ProtobufMapper.java:36)
	at it.auties.whatsapp.binary.BinarySocket$MessageHandler.handleProtocolMessage(BinarySocket.java:1405)
	at it.auties.whatsapp.binary.BinarySocket$MessageHandler.decodeMessage(BinarySocket.java:1303)
	at it.auties.whatsapp.binary.BinarySocket$MessageHandler.lambda$decode$35(BinarySocket.java:1275)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at it.auties.whatsapp.binary.BinarySocket$MessageHandler.decode(BinarySocket.java:1275)
	at it.auties.whatsapp.binary.BinarySocket$StreamHandler.digestMessage(BinarySocket.java:558)
	at it.auties.whatsapp.binary.BinarySocket$StreamHandler.digest(BinarySocket.java:545)
	at it.auties.whatsapp.binary.BinarySocket.handleNode(BinarySocket.java:206)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at it.auties.whatsapp.binary.BinarySocket.onBinary(BinarySocket.java:201)
	at jdk.internal.reflect.GeneratedMethodAccessor15.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.glassfish.tyrus.core.AnnotatedEndpoint.callMethod(AnnotatedEndpoint.java:553)
	at org.glassfish.tyrus.core.AnnotatedEndpoint$WholeHandler$1.onMessage(AnnotatedEndpoint.java:652)
	at org.glassfish.tyrus.core.TyrusSession.notifyMessageHandlers(TyrusSession.java:560)
	at org.glassfish.tyrus.core.TyrusEndpointWrapper.onMessage(TyrusEndpointWrapper.java:800)
	at org.glassfish.tyrus.core.TyrusWebSocket.onMessage(TyrusWebSocket.java:183)
	at org.glassfish.tyrus.core.frame.BinaryFrame.respond(BinaryFrame.java:70)
	at org.glassfish.tyrus.core.ProtocolHandler.process(ProtocolHandler.java:819)
	at org.glassfish.tyrus.client.TyrusClientEngine$TyrusReadHandler.handle(TyrusClientEngine.java:725)
	at org.glassfish.tyrus.container.grizzly.client.GrizzlyClientFilter$ProcessTask.execute(GrizzlyClientFilter.java:452)
	at org.glassfish.tyrus.container.grizzly.client.TaskProcessor.processTask(TaskProcessor.java:91)
	at org.glassfish.tyrus.container.grizzly.client.TaskProcessor.processTask(TaskProcessor.java:68)
	at org.glassfish.tyrus.container.grizzly.client.GrizzlyClientFilter.handleRead(GrizzlyClientFilter.java:248)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:88)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:246)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:178)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:118)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:96)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:51)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:510)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:82)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:83)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:34)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:101)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:535)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:515)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NullPointerException: syncType is marked non-null but is null
	at it.auties.whatsapp.model.sync.HistorySync$HistorySyncBuilder.syncType(HistorySync.java:21)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeSetAndReturn(MethodProperty.java:170)
	... 46 more

**Comment by Auties00** [comment]
I've investigated the bug a bit and it looks like the app patches pull system is not working as it should be. I've even looked at other implementations from other libraries, but they all don't do exactly what Whatsapp does. This inevitably creates a bad mac at some point. It fixes itself after you disconnect all your devices(may take a couple of tries), but I'm trying to reverse it perfectly from Whatsapp to have it work perfectly. I'll push an update asap

**Comment by Auties00** [comment]
If anyone wants to try the fix, use version 3.0.3

**Comment by Auties00** [comment]
Btw this is fixed in the latest version, it turns out that sending a request for presence updates before a certain point will make Whatsapp throw a bad Mac error
